services/src/handlers/layers.rs

```diff
@@ -455,3 +464,391 @@ async fn layer_handler<C: Context>(
+#[serde(rename_all = "camelCase")]
+pub struct AddLayerRequest {
+    pub layer: AddLayer,
+    #[schema(value_type = String, example = "05102bb3-a855-4a37-8a8a-30026a91fef1")]
+    pub collection_id: LayerCollectionId,
+}
+
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct AddLayerResponse {
+    pub layer_id: LayerId,
+}
+
+/// Add a new layer to a collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layers",
+    request_body = AddLayerRequest,
+    responses(
+        (status = 200, description = "OK", body = AddLayerResponse,
+            example = json!({
+                "layerId": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_layer<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add layers to their stuff
+    ctx: web::Data<C>,
+    request: web::Json<AddLayerRequest>,
+) -> Result<web::Json<AddLayerResponse>> {
+    let request = request.into_inner();
+
+    let collection_id = request.collection_id;
+    let add_layer = request.layer.validated()?;
+
+    let layer_id = ctx
+        .layer_db_ref()
+        .add_layer(add_layer, &collection_id)
+        .await?;
+
+    Ok(web::Json(AddLayerResponse { layer_id }))
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct AddCollectionRequest {
+    pub layer_collection: AddLayerCollection,
+    #[schema(value_type = String, example = "05102bb3-a855-4a37-8a8a-30026a91fef1")]
+    pub parent_id: LayerCollectionId,
+}
+
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct AddCollectionResponse {
+    pub layer_collection_id: LayerCollectionId,
+}
+
+/// Add a new collection to an existing collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layers/collections",
+    request_body = AddCollectionRequest,
+    responses(
+        (status = 200, description = "OK", body = AddCollectionResponse,
+            example = json!({
+                "layerCollectionId": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_collection<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add collections to their stuff
+    ctx: web::Data<C>,
+    request: web::Json<AddCollectionRequest>,
+) -> Result<web::Json<AddCollectionResponse>> {
+    let request = request.into_inner();
+
+    let parent_id = request.parent_id;
+    let add_collection = request.layer_collection.validated()?;
+
+    let layer_collection_id = ctx
+        .layer_db_ref()
+        .add_collection(add_collection, &parent_id)
+        .await?;
+
+    Ok(web::Json(AddCollectionResponse {
+        layer_collection_id,
+    }))
+}
+
+#[derive(Debug, Serialize, Deserialize, IntoParams)]
+struct RemoveCollectionFilter {
```

Comment:
I think a handler should either remove layers or collections. So maybe add another handler. 

- "DELETE /layers/collections/{collextionId}" to delete a collection
- "DELETE /layers/collections/{collectionId}/layers" to delete all layers
- "DELETE /layers/collections/{collectionId}/layers/{layerId}" to delete a single layer